homeassistant/components/fritzbox/light.py

```diff
@@ -61,6 +59,12 @@ def __init__(
         super().__init__(coordinator, ain, None)
         self._supported_hs: dict[int, list[int]] = {}
 
+        self._attr_supported_color_modes = {ColorMode.ONOFF}
```

Comment:
The onoff mode must be the only supported mode if present.

https://developers.home-assistant.io/docs/core/entity/light#color-modes

I'd use the helper to remove surplus modes.

https://github.com/home-assistant/core/blob/221f9615742f41307b17b7fda7a2ff6a7ccecd7b/homeassistant/components/light/__init__.py#L120

Reply:
> The onoff mode must be the only supported mode if present.

this is already ensured with the given code, since init with on/off mode, than overwrite the supported modes with a precedence for the color and color temp and if not suitable than the brightness mode (_if suitable_)

Reply:
or do you mean to do like in wiz light:
https://github.com/home-assistant/core/blob/2a8feda69116d0884965156ee78d444ab3803fdf/homeassistant/components/wiz/light.py#L82-L89